### ISSUE TITLE
Make the retry mechanism more configurable

### ISSUE THREAD

**Issue by Richard-Degenne** [googleapis/google-api-ruby-client#690]
Hi,

As you can see [here](https://github.com/google/google-api-ruby-client/blob/2a4efbf3b9c0e39255c574bfdf9ec57489857fcf/lib/google/apis/core/http_command.rb#L93-L96), it is currently possible to configure the number of retries through request options, but not other things such as the retry intervals, which are hard-coded, or [other `retriable` options](https://github.com/kamui/retriable#options).

Can something be done about that, or is it a policy to have those magic numbers as intervals?

**Comment by kented** [comment]
While uploading large files to Youtube I was receiving 'Google::Apis::TransmissionError: execution expired'. After some digging I learned that retriable has a default max_elapsed_time of 15 minutes. I was able to adjust this configuration globally to solve my problem. 

Would be great to have some sort of documentation for this underlying mechanism and/or expose some level of configuration directly. The [context](https://github.com/kamui/retriable#contexts) feature of retriable might be useful for this.
